script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,105 @@
+import fs from 'fs';
+import fetch from 'node-fetch';
+import path from 'path';
+import { argv } from 'yargs';
+import { LinkedArchiveTreeNode } from '../src/app/content/types';
+import { flattenArchiveTree } from '../src/app/content/utils';
+import archivedSlugs from '../src/archived-slugs.json';
+import { ARCHIVE_URL, REACT_APP_ARCHIVE_URL, REACT_APP_OS_WEB_API_URL } from '../src/config';
+import books from '../src/config.books.json';
+import createArchiveLoader from '../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../src/gateways/createOSWebLoader';
+
+interface ArchivedSlug {
+  pathname: string;
+  bookId: string;
+  pageId: string;
+}
+
+type ArchivedSlugs = ArchivedSlug[];
+
+const { book, newVersion } = argv as any as {
+  book: string
+  newVersion: string | number;
+};
+
+(global as any).fetch = fetch;
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const archivedSlugsPath = path.resolve(__dirname, '../src/archived-slugs.json');
+
+const archiveLoader = createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`);
+const osWebLoader = createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+async function processBooks() {
+  const booksData = Object.entries(books);
+  const bookToUpdateIndex = booksData.findIndex(([id]) => id === book);
+  const bookToUpdate = booksData[bookToUpdateIndex] as [string, { defaultVersion: string }] | undefined;
+
+  if (!bookToUpdate) {
+    console.error(`book ${book} not found`); // tslint:disable-line:no-console
```

Comment:
this script should also add new books, sorry i didn't think to mention that it was already handling that